crates/turbopack-css/src/chunk/mod.rs

```diff
@@ -69,6 +76,97 @@ impl CssChunkVc {
         }
         Ok(FileSystemPathOptionVc::cell(Some(current)))
     }
+
+    #[turbo_tasks::function]
+    async fn chunk_content(self) -> Result<CssChunkContentVc> {
+        let this = self.await?;
+        let chunk_path = self.path();
+        Ok(CssChunkContentVc::new(
+            this.main_entries,
+            this.context,
+            chunk_path,
+        ))
+    }
+}
+
+#[turbo_tasks::value]
+struct CssChunkContent {
+    main_entries: CssChunkPlaceablesVc,
+    context: ChunkingContextVc,
+    chunk_path: FileSystemPathVc,
+}
+
+#[turbo_tasks::value_impl]
+impl CssChunkContentVc {
+    #[turbo_tasks::function]
+    async fn new(
+        main_entries: CssChunkPlaceablesVc,
+        context: ChunkingContextVc,
+        chunk_path: FileSystemPathVc,
+    ) -> Result<Self> {
+        Ok(CssChunkContent {
+            main_entries,
+            context,
+            chunk_path,
+        }
+        .cell())
+    }
+
+    #[turbo_tasks::function]
+    async fn code(self) -> Result<CodeVc> {
+        let this = self.await?;
+        let chunk_name = this.chunk_path.to_string();
+
+        let mut external_imports = IndexSet::new();
+        let mut codes = Vec::new();
+        for entry in this.main_entries.await?.iter() {
+            let entry_placeable = CssChunkPlaceableVc::cast_from(entry);
+            let entry_item = entry_placeable.as_chunk_item(this.context);
+            let expanded = expand_imports(entry_item).await?;
+
+            for external_import in &expanded.external_imports {
+                external_imports.insert(*external_import);
```

Comment:
The Set no longer deduplicates duplicate imports now, since it no longer stores the String, but the Vc.

But for parallelization reasons it cool to move the `await?` to the later location.